conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_3/section_3_2.rs

```diff
@@ -82,6 +82,36 @@ fn on_do_0_query_strips_dnssec_records_even_if_it_cached_a_previous_do_1_query()
     Ok(())
 }
 
+// this ensures that in the presence of a cached entry (answer), the dnssec records (answer+rrsig) are still
+// returned as per the RFC
+#[test]
+fn on_do_1_query_return_dnssec_records_even_if_it_cached_a_previous_do_0_query() -> Result<()> {
+    let network = &Network::new()?;
+    let ns = NameServer::new(&dns_test::PEER, FQDN::ROOT, network)?
+        .sign()?
+        .start()?;
+    let resolver = Resolver::new(network, ns.root_hint()).start(&dns_test::SUBJECT)?;
+
+    let client = Client::new(network)?;
+    let settings = *DigSettings::default().recurse();
+    let resolver_addr = resolver.ipv4_addr();
+    let ans = client.dig(settings, resolver_addr, RecordType::SOA, &FQDN::ROOT)?;
+
+    let [answer] = ans.answer.try_into().unwrap();
+
+    assert!(matches!(answer, Record::SOA(_)));
+
+    let settings = *DigSettings::default().dnssec().recurse();
+    let ans = client.dig(settings, resolver_addr, RecordType::SOA, &FQDN::ROOT)?;
```

Comment:
this is meant to be a test of the caching behavior (+) so it should check that no network communication with the nameservers happens on the second query using the `eavesdrop` API. see #2244 for an example of that.

the current assertions should be preserved.

 (+) for that reason, I would also put this test inside module `section_4_5` which is created in #2244